google/cloud/ndb/model.py

```diff
@@ -2385,6 +2395,14 @@ def _db_set_uncompressed_meaning(self, p):
         """
         raise exceptions.NoLongerImplementedError()
 
+    def _prepare_for_put(self, entity):
+        """Set up meanings for backwards compatibility."""
+        if self._compressed:
+            entity._meanings[self._name] = (
+                _MEANING_URI_COMPRESSED,
+                self._to_base_type(self._get_user_value(entity)),
+            )
+
```

Comment:
else:
    entity._meanings[self._name] = (???, value)

Reply:
Not so sure about this, since we'd have to know the meaning values for everything and I'm not sure `_meanings` are set up for other types in the backend.

Reply:
My point here is just that we need to tell the database when it's not compressed, as well. If ``_compressed`` is ``False``, but it was previously written compressed, then we don't want to pass a compressed meaning back to the Datastore. Maybe the move is actually:

~~~~
entity._meanings.pop(self._name, None)
~~~~

Or, maybe the move is we don't have an ``entity._meanings`` attribute, and generate meanings from scratch when serializing, maybe by overriding ``_to_datastore``. 

I'm leaning towards that last option, myself, since we don't really need the meanings in NDB except when serializing and deserializing and we don't want to use the meanings from deserializing when reserializing.